src/OrleansSQLUtils/Storage/AdoNetConfigurationExtensions.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Orleans.Runtime.Configuration;
+using Orleans.Storage;
+using System;
+using System.Collections.Generic;
+
+
+namespace OrleansSQLUtils.Storage
+{
+    /// <summary>
+    /// The format in which the data will serialized to the storage.
+    /// </summary>
+    public enum AdoNetSerializationFormat
+    {
+        /// <summary>
+        /// The native Orleans binary serialization format.
+        /// </summary>
+        OrleansNativeBinary = 1,
+
+        /// <summary>
+        /// XML.
+        /// </summary>
+        Xml                 = 2,
+
+        /// <summary>
+        /// JSON.
+        /// </summary>
+        Json                = 3
+    }
+
+
+    /// <summary>
+    /// Extension methods for configuration classes specific to <see cref="OrleansSQLUtils"/>.
+    /// </summary>
+    public static class AdoNetConfigurationExtensions
+    {
+        /// <summary>
+        /// Adds a storage provider of type <see cref="AdoNetStorageProvider"/>.
+        /// </summary>
+        /// <param name="config">The cluster configuration object to add provider to.</param>
+        /// <param name="providerName">The provider name.</param>
+        /// <param name="connectionString">The ADO.NET storage connection string. If none is provided, it will use the same as in the Globals configuration.</param>
+        /// <param name="serializationFormat">The format in which the data will serialized to the storage.</param>
+        public static void AddAdoNetStorageProvider(
+            this ClusterConfiguration config,
+            string providerName = "AdoNetStorage",
+            string connectionString = null,
+            AdoNetSerializationFormat serializationFormat = AdoNetSerializationFormat.OrleansNativeBinary)
```

Comment:
do we really want to have Orleans native as the default? I would really discourage this if there is some alternative that works fine. Orleans serializer is not version tolerant, which is a very bad thing for stuff persisted in durable storage. I think in newer providers (ie: azure blob storage) we even stopped providing Orleans binary serializer as an option.